lib/workers/repository/process/lookup/filter.ts

```diff
@@ -37,25 +37,27 @@ export function filterVersions(
   }
 
   // Leave only versions greater than current
-  let filteredVersions = releases.filter(
-    (v) =>
-      versioning.isVersion(v.version) &&
-      versioning.isGreaterThan(v.version, currentVersion),
+  let filteredReleases = releases.filter(
+    (r) =>
+      versioning.isVersion(r.version) &&
+      versioning.isGreaterThan(r.version, currentVersion),
   );
 
-  // Don't upgrade from non-deprecated to deprecated
-  const fromRelease = releases.find(
-    (release) => release.version === currentVersion,
+  const currentRelease = releases.find(
+    (r) =>
+      versioning.isValid(r.version) &&
+      versioning.isVersion(r.version) &&
+      versioning.isValid(currentVersion) &&
+      versioning.isVersion(currentVersion) &&
+      versioning.equals(r.version, currentVersion),
```

Comment:
This modification meant to prevent mismatches like leading `v` character

Reply:
If there's an intentional change of functionality, please add a test

Reply:
Actually, it's for separate PR